platform/security/session-management-endpoint/src/main/java/org/codice/ddf/security/session/management/impl/SessionManagementEndpointImpl.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.session.management.impl;
+
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.assertion.impl.SecurityAssertionImpl;
+import ddf.security.common.SecurityTokenHolder;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import javax.ws.rs.core.Response;
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.codice.ddf.security.handler.api.SAMLAuthenticationToken;
+import org.codice.ddf.security.session.management.service.SessionManagementCommonService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SessionManagementEndpointImpl implements SessionManagementCommonService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SessionManagementEndpointImpl.class);
+
+  private SecurityManager securityManager;
+
+  private Clock clock = Clock.systemUTC();
+
+  private String rootContext;
+
+  public InputStream getExpiry(HttpServletRequest request) {
```

Comment:
❓ Why are the signatures of these public methods `InputStream`? You could just return the UTF-8 encoded strings directly.